testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveRedirectionToHudi.java

```diff
@@ -0,0 +1,438 @@
+                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): line 1:1: Table " + hiveTableName + " is redirected to " + hudiTableName + " and REVOKE is not supported with table redirections");
+
+        onHudi().executeQuery("DROP TABLE " + schemaTableName);
+    }
+
+    @Test(groups = {HIVE_HUDI_REDIRECTIONS, PROFILE_SPECIFIC_TESTS})
+    public void testSetTableAuthorization()
+    {
+        String tableName = "hudi_set_table_authorization_" + randomNameSuffix();
+        String schemaTableName = schemaTableName("default", tableName);
+        String hudiTableName = trinoTableName("hudi", schemaTableName);
+        String hiveTableName = trinoTableName("hive", schemaTableName);
+
+        createHudiCowTable(schemaTableName, false);
+
+        assertQueryFailure(() -> onTrino().executeQuery("ALTER TABLE " + hiveTableName + " SET AUTHORIZATION ROLE PUBLIC"))
+                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): line 1:1: Table " + hiveTableName + " is redirected to " + hudiTableName + " and SET TABLE AUTHORIZATION is not supported with table redirections");
+
+        onHudi().executeQuery("DROP TABLE " + schemaTableName);
+    }
+
+    @Test(groups = {HIVE_HUDI_REDIRECTIONS, PROFILE_SPECIFIC_TESTS})
+    public void testDeny()
+    {
+        String tableName = "hudi_deny_" + randomNameSuffix();
+        String schemaTableName = schemaTableName("default", tableName);
+        String hudiTableName = trinoTableName("hudi", schemaTableName);
+        String hiveTableName = trinoTableName("hive", schemaTableName);
+
+        createHudiCowTable(schemaTableName, false);
+
+        assertQueryFailure(() -> onTrino().executeQuery("DENY DELETE ON " + hiveTableName + " TO ROLE PUBLIC"))
+                .hasMessageMatching("\\QQuery failed (#\\E\\S+\\Q): line 1:1: Table " + hiveTableName + " is redirected to " + hudiTableName + " and DENY is not supported with table redirections");
+
+        onHudi().executeQuery("DROP TABLE " + schemaTableName);
+    }
+
+    private static String schemaTableName(String schema, String tableName)
+    {
+        return "%s.%s".formatted(schema, tableName);
+    }
+
+    private static String trinoTableName(String catalog, String schemaTableName)
+    {
+        return "%s.%s".formatted(catalog, schemaTableName);
+    }
+
+    private static void assertResultsEqual(QueryResult first, QueryResult second)
+    {
+        assertThat(first).containsOnly(second.rows().stream()
+                .map(QueryAssert.Row::new)
+                .collect(toImmutableList()));
+
+        // just for symmetry
+        assertThat(second).containsOnly(first.rows().stream()
+                .map(QueryAssert.Row::new)
+                .collect(toImmutableList()));
+    }
+
+    private void createHudiCowTable(String tableName, boolean partitioned)
+    {
+        createHudiTable(tableName, HUDI_TABLE_TYPE_COPY_ON_WRITE, partitioned);
+    }
+
+    private void createHudiMorTable(String tableName, boolean partitioned)
```

Comment:
`MOR` table is only tested in `testMerge` that verifies failure. Could you add another two test methods for reading MOR partitioned & non-partitioned tables? or adding data provider to existing methods might be enough.

Reply:
fixed